packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -93,7 +94,7 @@ export async function makeWallet(
     if (zoeKind === 'keywords') {
       // Payments are made for the keywords in proposal.give.
       payment = {};
-      await Promise.all(
+      await allComparable(
```

Comment:
Actually, the `Promise.all` was working, it's just that the `payment[keyword]` was still a promise at the end of it.  Here's what we could do if wanting to avoid changing line 100.

```suggestion
      const paymentPs = {};
      Object.entries(proposal.give || {}).map(([keyword, amount]) => {
        const purse = purses[keyword];
        if (purse) {
          paymentPs[keyword] = E(purse).withdraw(amount);
        }
      });
      payment = await allComparable(paymentPs);
```

Reply:
I slightly prefer avoiding allComparable:
```js
      // Payments are made for the keywords in proposal.give.
      payment = {};
      await Promise.all(
        Object.entries(proposal.give || {}).map(([keyword, amount]) => {
          const purse = purses[keyword];
          if (purse) {
            return E(purse)
              .withdraw(amount)
              .then(pmt => (payment[keyword] = pmt));
          }
          return undefined;
        }),
      );
```

Reply:
>  it's just that the payment[keyword] was still a promise at the end of it.

That's what I would expect. I think `allComparable` would ensure that payment[keyword] is turned into a payment, right? I'm confused about why we would need to change the current change.

Reply:
FTR, the problem was that the old code was capturing the promise in the payment[keyword] record, not its resolution.